src/utils/keystore.ts

```diff
@@ -47,15 +59,21 @@ export async function deriveKeyUsingArgon2id (pass, salt, options) {
 }
 
 // CRYPTO PART
-const CRYPTO_FUNCTIONS = {
+const CRYPTO_FUNCTIONS: {[key: string]: any} = {
   'xsalsa20-poly1305': { encrypt: encryptXsalsa20Poly1305, decrypt: decryptXsalsa20Poly1305 }
 }
```

Comment:
```suggestion
interface CryptoAlgorithm {
  encrypt:
    (opts: {
      plaintext: Buffer|string|Uint8Array,
      key: Buffer|string|Uint8Array,
      nonce: Buffer|string|Uint8Array
    }) => Uint8Array | Buffer
  decrypt: (opts: {
    ciphertext: Buffer|string|Uint8Array,
    key: Buffer|string|Uint8Array,
    nonce: Buffer|string|Uint8Array}) => Uint8Array | Buffer
}

// CRYPTO PART
const CRYPTO_FUNCTIONS: {[key: string]: CryptoAlgorithm} = {
  'xsalsa20-poly1305': { encrypt: encryptXsalsa20Poly1305, decrypt: decryptXsalsa20Poly1305 }
}
```

What would your opinion be about defining an interface for the algorithms? Even if that'd create the necessity of a new interface along with the addition of Uint8Array to several `string | Buffer` 

:bulb:  **Note**: If you proceed with this approach, function `encrypt` will need to return `Buffer | Uint8Array`